h2o-algos/src/main/java/hex/tree/SharedTree.java

```diff
@@ -427,6 +441,9 @@ protected final void scoreAndBuildTrees(boolean oob) {
           break; // If timed out, do the final scoring
         }
         if (stop_requested()) throw new Job.JobCancelledException();
+        if (tid == _ntrees - 1 && _coordinator != null) {
+          _coordinator = _coordinator.updateParameters();
+        }
```

Comment:
the condition on `tid` is important, basically it means - as longs as you you know you have enough trees to build don't call the coordinator and don't ask for new parameters

only if I reach the last updated number of trees, I ask for a new parameter - I also know that I will either get a new increased number or nothing will happen and the coordinator will be null (in that case the _ntrees can increase but doesn't have to)

Reply:
Yes, I understand now, it's because `updateParameters` may change the `_ntrees` so the loop may keep running after this  statement.

It's just my habits: when I see a `for` loop, I don't expect the condition to change, I always use a `while` loop otherwise.